Update installation doc for IntelliJ dynamic

ches opened the pull request:
Opening a PR to claim #1396, someone who's able can assign that issue to me.

I've pushed a few small doc updates _not_ including updating the installation section for IntelliJ Scala Plugin with scalafmt-dynamic support yet—I'll need to return to it later today but I have it underway and will update the branch.

- [x] [Generate 2.0.0-RC6 changelog](https://scalameta.org/scalafmt/docs/contributing-scalafmt.html#releasing) while I'm updating the site

er1c commented:
So far looks good, if you're already on the branch, can you look into this: https://github.com/scalameta/scalafmt/issues/1392

olafurpg commented:
Thank you for this contribution! Changes so far look good, happy to merge once the IntelliJ Scala section has been updated. Make sure to mention the latest 2019 version is required

ches commented:
Okay:

- Updated installation doc for IntelliJ dynamic
- Cherry-picked the #1398 change here
- Fixed #1392 
- Updated changelog for 2.0.0-RC6. Note: the changelog generator did not usefully work, since RC5 it seems there have been only squash/rebase merges and the tool couldn't figure those out:

   ```
   Warning: PR 1395 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
   Warning: PR 1381 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
   Warning: PR 1380 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
   Warning: PR 1379 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
   Warning: PR 1378 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
   Warning: PR 1377 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
   ```

   I manually added the changelog entries this time, pruned out a few merged PRs that I don't think are very meaningful to users. We might want to opt for always creating merge commits, or seeking an alternative solution for automating changelog.
